src/Impostor.Server/Net/Inner/Objects/InnerGameData.TaskInfo.cs

```diff
@@ -20,9 +23,9 @@ public void Serialize(IMessageWriter writer)
 
             public void Deserialize(IMessageReader reader)
             {
-                this.Type = (TaskTypes)reader.ReadPackedUInt32();
+                TaskIndex = reader.ReadPackedUInt32();
                 this.Complete = reader.ReadBoolean();
             }
```

Comment:
The deserialize method doesn't assign a tasktype, so it might be missing in some cases. Maybe get the tasktype from the player info in here somehow?

Reply:
No, because it is set for the player info by RPC set tasks, where it sets it here as well (RPC set tasks and RPC game data is sent together). For the API, a SetTasks API call should update the player info tasks and the task indexes itself.